lib/rmg/TreasurePlacer.h

```diff
@@ -24,14 +24,12 @@ struct ObjectInfo
 	std::shared_ptr<const ObjectTemplate> templ;
 	ui32 value = 0;
 	ui16 probability = 0;
-	ui32 maxPerZone = -1;
+	ui32 maxPerZone = 1;
```

Comment:
Now I like in-place initialization even less. Because such cases WILL appear over time making value from header even less useful than before

Reply:
@IvanSavenko but is there any point to make in-place init to -1 and in one and only constructor initialize field to 1?

Reply:
That's the point. Or lack of it.
Since it is now possible to initialize member in two places, we will eventually end up in such scenario where in-place init and in-constructor init set some field to different values, leading to misleading information in header.

Reply:
I think if we have only one constructor without parameters, which initializes fields to primitive types and nullptr, we can use in-place init and trivial constructor. You think than hide implementation details, but define a non-trivial constructor with initializers is better?

Reply:
In this particular case - yeah. Not even sure how this code ended up like this.

Such approach *is* nice for small c-like structs that have no methods of their own and have definition less than ~10 lines in size.
In such cases (including this code) yeah, I think it is fine to use inplace init. 

But for anything larger and more complex than that - probably better to avoid it, specifically to prevent such cases.

Reply:
I tried to force removal of constructors (it is almost all hand changes) only in cases when struct have `serialize()` method and follow netpacks pattern. I think there is a good change. In big classes - not so much.

Reply:
I think in this place I can save this change. @IvanSavenko do you agree?
In-place init will be difficult to read if we have more than 1 default constructor, I agree with you in this point.

Reply:
Yes, here change is fine, my comment was mostly to highlight potential issue with using in-place init on large classes that are too big for all relevant code to be visible at glance.